src/wolff_producer.erl

```diff
@@ -58,12 +59,12 @@
                     max_linger_ms => non_neg_integer(),
                     max_send_ahead => non_neg_integer(),
                     compression => kpro:compress_option(),
-                    drop_if_highmem => boolean()
+                    drop_if_highmem => boolean(),
+                    telemetry_meta_data => term()
```

Comment:
I think this can only be a map, according to what `telemetry:execute/3` expects.

```suggestion
                    telemetry_meta_data => map()
```

Reply:
Correct! Fixed!